### PR TITLE
osd/scrub: initial set of performance counters

### PR DESCRIPTION
**Pull request by ronen-fr**
The added counters are (mostly) duplicated for 4 labels (shallow/deep times and EC vs. replicated).

### CODE REVIEW

File: src/osd/scrubber/scrub_reservations.cc [hunk 1]
```diff
@@ -88,9 +95,34 @@ void ReplicaReservations::discard_remote_reservations()
...
+  auto logged_duration = ScrubClock::now() - m_process_started_at.value();
+  m_perf_set.tinc(scrbcnt_resrv_successful_elapsed, logged_duration);
+  m_perf_set.inc(scrbcnt_resrv_success);
+  m_osds->logger->hinc(
+      l_osd_scrub_reservation_dur_hist, m_sorted_secondaries.size(),
+      logged_duration.count());
+  m_process_started_at.reset();
+}
+
+void ReplicaReservations::log_failure_and_duration(int reason_key)
```
Comment (NitzanMordhai) [inline src/osd/scrubber/scrub_reservations.cc:109]: I'm not sure about reason_key, maybe name it differently? its the counter increas

**Review by NitzanMordhai — COMMENTED** [review]
Looks good, i left one comment over

File: src/osd/scrubber/scrub_reservations.cc (hunk 1 above)
Comment (ronen-fr) [reply src/osd/scrubber/scrub_reservations.cc:109]: It's the 'failure reason counter'.
I'll change to 'failure_cause_counter'

**Comment by ronen-fr** [comment]
> Looks good, i left one comment over

Thanks!

**Review by NitzanMordhai — APPROVED** [review]
LGTM

**Comment by ronen-fr** [comment]
Merging based on https://pulpito.ceph.com/rfriedma-2023-12-20_08:40:50-rados-wip-rf-pcnt3-distro-default-smithi/
This is a full set of Rados tests. 22 failures (16f, 6dead).
All failures (apart from the 3 deads that are reimaging/infra issues) match 'main':
see https://pulpito.ceph.com/yuriw-2023-12-11_16:41:51-rados-wip-yuri3-testing-2023-12-07-0727-distro-default-smithi/

**Pull request merged**